plugins/filter_kubernetes/kube_meta.c

```diff
@@ -87,14 +89,144 @@ static int file_to_buffer(const char *path,
     return 0;
 }
 
+/* Run command to get Kubernetes authorization token */
+static int get_token_with_command(const char *command, 
+                                  char **out_buf, size_t *out_size)
+{
+    FILE *fp;
+    char buf[FLB_KUBE_TOKEN_BUF_SIZE];
+    char *temp;
+    char *res;
+    size_t size = 0;
+    size_t len = 0;
+
+    fp = popen(command, "r");
+    if (fp == NULL) {
+        return -1;
+    }
+
+    res = flb_calloc(1, FLB_KUBE_TOKEN_BUF_SIZE);
+    if (!res) {
+        flb_errno();
+        fclose(fp);
+        return -1;
+    }
+    
+    while (fgets(buf, sizeof(buf), fp) != NULL) {
+        len = strlen(buf);
+        if (len >= FLB_KUBE_TOKEN_BUF_SIZE - 1) {
+            temp = flb_realloc(res, FLB_KUBE_TOKEN_BUF_SIZE * 2);
```

Comment:
`FLB_KUBE_TOKEN_BUF_SIZE` is a constant. So I think this code doesn't work. You would never re-alloc more than `2 * FLB_KUBE_TOKEN_BUF_SIZE`.

Reply:
I see. I will change it to `size + FLB_KUBE_TOKEN_BUF_SIZE`